_shared/local_extensions.py

```diff
@@ -110,11 +113,81 @@ def include_exists(self, context, path):
+    def include_json(self, context, path):
+        """Return the project's .cookiecutter/includes/{path} data or None.
+
+        Return the contents of the project's .cookiecutter/includes/{path} JSON
+        file deserialized into a Python object.
+
+        Return None if the project has no .cookiecutter/includes/{path} file.
+        """
+        return json.load(self._open(context, path))
+
+    @pass_context
+    def supervisor_settings(self, context):
+        """Return the settings from the project's supervisor.json file.
+
+        Or return the default supervisor settings.
+        """
+        # Are we rendering the production supervisord.conf or the development environment's supervisor-dev.conf?
+        if context.name.endswith("supervisord-dev.conf"):
+            dev = True
+        else:
+            assert context.name.endswith("supervisord.conf")
+            dev = False
+
+        try:
+            settings = self.include_json(context, "supervisor.json")
+        except FileNotFoundError:
+            settings = {}
+
+        if not settings.get("programs"):
+            if context["cookiecutter"]["_directory"] == "pyramid-app":
+                settings["programs"] = {"web": {"settings": {}}}
+                if dev:
+                    settings["programs"]["web"]["settings"][
+                        "command"
+                    ] = f"gunicorn --bind :{context['cookiecutter']['port']} --workers 1 --reload --timeout 0 --paste conf/development.ini"
+                else:
+                    settings["programs"]["web"]["settings"][
+                        "command"
+                    ] = f"newrelic-admin run-program gunicorn --paste conf/production.ini --bind 0.0.0.0:{context['cookiecutter']['port']}"
+            else:
+                settings["programs"] = {
+                    "app": {
+                        "settings": {
+                            "command": f"python3 -m {context['cookiecutter']['package_name']}",
+                        }
+                    }
+                }
+
+        for program in settings["programs"].values():
+            program.setdefault("dev", True)
+            program.setdefault("prod", True)
+            program.setdefault("settings", {})
+            program["settings"].setdefault("stdout_events_enabled", "true")
+            program["settings"].setdefault("stderr_events_enabled", "true")
+            if dev:
+                program["settings"].setdefault("stopsignal", "KILL")
+                program["settings"].setdefault("stopasgroup", "true")
+            else:
+                program["settings"].setdefault("stdout_logfile", "NONE")
+                program["settings"].setdefault("stderr_logfile", "NONE")
+
+        settings["programs"] = {
+            program_name: program_settings
+            for program_name, program_settings in settings["programs"].items()
+            if program_settings["dev" if dev else "prod"]
+        }
+
+        return settings
```

Comment:
There's two reasons why this Supervisor-specific method has to be implemented in Python:

1. It's a total pain to write this logic (setting default values etc) in Jinja. Ends up really ugly
2. With cookiecutter there's no way to shared template code or macros between files. But this Python function is available to all templates